### PR TITLE
Gamelab palette & dashboard integration

### PR DESCRIPTION
**Pull request by cpirich**
- Added a palette/toolbox for Game Lab (expressed in `codeFunctions` and `dropletConfig`)
- Extended the set of event functions that we automatically bind to for p5 to include keyboard and mouse events. In the process, threw out the old use of `eventHandlers` from playlab and repurposed it.
- Added some missing skin files (sounds, images) needed to expose gamelab in the dashboard
- Expose gamelab in the dashboard with a project level. Behavior is similar to applab, but for now, we're not requiring that you be signed in. This may change as we integrate the data APIs.
- Minimal support added to support level builder with gamelab levels
- Changed how the project system grabs the `getCode` function - instead of referencing the `Applab` global, use `appOptions.getCode`, which is now populated in `studioApp.init` (Note that this doesn't remove the use of the `Applab` global for `getHtml`)

### CODE REVIEW

File: apps/src/StudioApp.js [hunk 1]
```diff
@@ -249,6 +249,10 @@ StudioApp.prototype.init = function(config) {
     config = {};
   }
 
+  config.getCode = function () {
+    return this.getCode();
+  }.bind(this);
```
Comment (islemaster) [inline apps/src/StudioApp.js]: Nit: You could do this sans one stack layer.

``` javascript
config.getCode = this.getCode.bind(this);
```

File: apps/src/gamelab/GameLab.js [hunk 2]
```diff
@@ -411,9 +360,14 @@ GameLab.prototype.execute = function() {
     if (!this.JSInterpreter.initialized()) {
       return;
     }
-    this.drawFunc = this.JSInterpreter.findGlobalFunction('draw');
-    this.setupFunc = this.JSInterpreter.findGlobalFunction('setup');
-    this.mousePressedFunc = this.JSInterpreter.findGlobalFunction('mousePressed');
+
+    ['draw', 'setup'].concat(this.p5eventNames).forEach(function (eventName) {
```
Comment (islemaster) [inline apps/src/gamelab/GameLab.js]: Please de-dup with `['draw', 'setup']` literal on line 36.

Comment (islemaster) [inline apps/src/gamelab/GameLab.js]: What am I saying?  Replace `['draw', 'setup'].concat(this.p5eventNames)` with `this.p5specialFunctions`.

File: code-studio/src/js/initApp/initApp.js [hunk 3]
```diff
@@ -169,8 +169,8 @@ window.apps = {
         // If we're readOnly, source hasn't changed at all
         source = Blockly.mainBlockSpace.isReadOnly() ? currentLevelSource :
           Blockly.Xml.domToText(Blockly.Xml.blockSpaceToDom(Blockly.mainBlockSpace));
-      } else {
-        source = window.Applab && Applab.getCode();
+      } else if (appOptions.getCode) {
+        source = appOptions.getCode();
```
Comment (islemaster) [inline code-studio/src/js/initApp/initApp.js:173]: Unrelated to this PR: Did building `code-studio` to test this stuff give you any trouble?

**Comment by islemaster** [comment]
This rocks!  Nice code cleanup around the events.  LGTM.

File: code-studio/src/js/initApp/initApp.js (hunk 3 above)
Comment (cpirich) [inline code-studio/src/js/initApp/initApp.js:173]: Took me a little while to figure it out. I still have a shared directory which I think I should be ignoring. And I edited generated code a few times before I realized how it all worked!

File: apps/src/StudioApp.js (hunk 1 above)
Comment (cpirich) [inline apps/src/StudioApp.js]: of course!

File: code-studio/src/js/initApp/initApp.js (hunk 3 above)
Comment (islemaster) [inline code-studio/src/js/initApp/initApp.js:173]: Oops!  Sorry to hear that.  The shared directory is still around, but the shared/js package moved into code-studio.  Take a look at [the README for code-studio](https://github.com/code-dot-org/code-dot-org/tree/staging/code-studio) or ask me if you have any questions.

**Pull request merged**